rvv-intrinsic-generator/Makefile

```diff
@@ -0,0 +1,263 @@
+###############################################################################
+# Copyright 2022 SiFive Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+###############################################################################
+# Utilities
+###############################################################################
+# Check that given variables are set and all have non-empty values,
+# die with an error otherwise.
+#
+# Params:
+# 1. Variable name(s) to test.
+# 2. (optional) Error message to print.
+check_defined = \
+		$(strip $(foreach 1,$1, \
+				$(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+		$(if $(value $1),, \
+			$(error Undefined $1$(if $2, ($2))))
+
+# Replace softfloat float-point types with LLVM compatible floating-point types
+replace_float = \
+		sed -i 's/float16_t/_Float16/g' $(1)/*; \
+		sed -i 's/float32_t/float/g' $(1)/*; \
+		sed -i 's/float64_t/double/g' $(1)/*
+
+###############################################################################
+# Variables
+###############################################################################
+# Path to the rvv_intrinsic_gen folder
+RVV_INTRINSIC_GEN_PATH := $(CURDIR)/rvv_intrinsic_gen
+# Vendor specific folder
+VENDOR_PATH ?=
+# Absolute vendor path
+ABS_VENDOR_PATH := $(abspath $(VENDOR_PATH))
+# Python paths for module import
+PYTHONPATHS = $(RVV_INTRINSIC_GEN_PATH):$(ABS_VENDOR_PATH)
+# Requires Python3.6 or higher
+PY3 := PYTHONPATH=$$PYTHONPATH:$(PYTHONPATHS) python3
+# Main entry script of the generator
+MAIN := rvv_intrinsic_gen.main
+# Main output directory is default to golden
+OUTPUT_DIR := golden
+# Derives output directory for each set of intrinsics
+# Output directory for non-policy intrinsics
+DIR := $(CURDIR)/$(OUTPUT_DIR)
```

Comment:
Do not append `$(CURDIR)`, otherwise OUTPUT_DIR can't be absolute path, I would suggest use `$(abspath $(OUTPUT_DIR))` here is you would like to use full path for DIR